packages/dds/task-manager/src/taskManager.ts

```diff
@@ -312,25 +317,69 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         return lockAcquireP;
     }
 
-    public abandon(taskId: string) {
-        if (!this.connected) {
-            throw new Error(`Attempted to abandon in disconnected state: ${taskId}`);
+    public subscribeToTask(taskId: string) {
+        if (this.subscribed(taskId)) {
+            return;
         }
 
-        // Nothing to do if we're not at least trying to get the lock.
-        if (!this.queued(taskId)) {
-            return;
+        const submitVolunteerOp = () => {
+            this.submitVolunteerOp(taskId);
+        };
+
+        const disconnectHandler = () => {
+            // Wait to be connected again and then re-submit volunteer op
+            this.connectionWatcher.once("connect", submitVolunteerOp);
+        };
+
+        const checkIfAbandoned = (eventTaskId: string) => {
+            if (eventTaskId !== taskId) {
+                return;
+            }
+
+            this.abandonWatcher.off("abandon", checkIfAbandoned);
+            this.connectionWatcher.off("disconnect", disconnectHandler);
+            this.connectionWatcher.off("connect", submitVolunteerOp);
+
+            this.subscribedTasks.delete(taskId);
+        };
+
+        this.abandonWatcher.on("abandon", checkIfAbandoned);
+        this.connectionWatcher.on("disconnect", disconnectHandler);
+
+        if (!this.connected) {
+            disconnectHandler();
+        } else if (!this.assigned(taskId) && !this.queued(taskId)) {
+            // TODO simulate auto-ack in detached scenario
+            submitVolunteerOp();
         }
-        // TODO simulate auto-ack in detached scenario
-        if (!this.isAttached()) {
-            return;
+        this.subscribedTasks.add(taskId);
+    }
+
+    public abandon(taskId: string) {
+        // Always allow abandon if the client is subscribed to allow clients to unsubscribe while disconnected.
+        // Otherwise, we should check to make sure the client is both connected queued for the task before sending an
+        // abandon op.
+        if (!this.subscribed(taskId)) {
+            if (!this.connected) {
+                throw new Error(`Attempted to abandon in disconnected state: ${taskId}`);
+            }
+
+            // Nothing to do if we're not at least trying to get the lock.
+            if (!this.queued(taskId)) {
+                return;
+            }
+
+            // TODO simulate auto-ack in detached scenario
+            if (!this.isAttached()) {
+                return;
+            }
         }
 
         this.submitAbandonOp(taskId);
         this.abandonWatcher.emit("abandon", taskId);
```

Comment:
This will still attempt to submit an abandon op when subscribed and offline.

I think one thing you can do to simplify is remove the throw when offline (this was always kind of an over-aggressive failfast, moreso now that calling abandon while offline is now a valid scenario for abandoning subscriptions)

So maybe this looks something like

```suggestion
        if (!this.subscribed(taskId) && !this.queued(taskId)) {
            // Nothing to do
            return;
        }

        // TODO simulate auto-ack in detached scenario
        if (!this.isAttached()) {
            return;
        }

        // If we're subscribed but not queued, we don't need to submit an abandon op (probably offline)
        if (this.queued(taskId)) {
            this.submitAbandonOp(taskId);
        }
        this.abandonWatcher.emit("abandon", taskId);
```